src/interfaces/alteration-interfaces/alteration-interface.ts

```diff
@@ -26,9 +25,8 @@ export interface AlterationIF extends CoopAlterationIF {
     identifier: string
     legalType: CorpTypeCd
   }
-  /** This is different from bcrs-shared-components/interfaces/name-request-interface.ts
-   * and should be refactored in the future. */
-  nameRequest?: NameRequestIF
+  // intersection type so we can save original NR + properties needed by Legal API:
+  nameRequest?: NameRequestIF & { legalName: string, nrNumber?: string }
```

Comment:
Nice! First time I see this syntax. Cool.

Reply:
🎉 

I knew this was possible but I had to look up how to do it! This is a good way to keep a tight reign on what properties (and types) an object contains.

Background: in the early days, we'd include a property like this in our objects to allow undeclared properties. But it was a bad idea so I remove all of these a long time ago.
https://github.com/bcgov/business-filings-ui/blob/b8bed27524026426ce810061952a048ab5f51fe7/src/interfaces/filing-interface.ts#L14

Reply:
I see, OK. Will keep this in my head moving forward. Would be cool to use it myself down the line. As a Padawan, there's always something new to learn from the Jedi Master 😆

Reply:
https://twitter.com/KingOnTheSite/status/1465636874844790785

Reply:
Hahahahahahaha, love this so much